gtk4/src/subclass/widget.rs

```diff
@@ -53,19 +198,317 @@ impl<T: WidgetImpl> WidgetImplExt for T {
+            from_glib(f(widget.unsafe_cast_ref::<Widget>().to_glib_none().0))
+        }
+    }
+
+    fn parent_grab_focus(&self, widget: &Self::Type) -> bool {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkWidgetClass;
+            if let Some(f) = (*parent_class).grab_focus {
+                from_glib(f(widget.unsafe_cast_ref::<Widget>().to_glib_none().0))
+            } else {
+                false
+            }
+        }
+    }
+
+    fn parent_hide(&self, widget: &Self::Type) {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkWidgetClass;
+            if let Some(f) = (*parent_class).hide {
+                f(widget.unsafe_cast_ref::<Widget>().to_glib_none().0)
+            }
+        }
+    }
+
+    fn parent_keynav_failed(&self, widget: &Self::Type, direction_type: DirectionType) -> bool {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkWidgetClass;
+            if let Some(f) = (*parent_class).keynav_failed {
+                f(
+                    widget.unsafe_cast_ref::<Widget>().to_glib_none().0,
+                    direction_type.to_glib(),
+                ) != 0
+            } else {
+                false
+            }
+        }
+    }
+
+    fn parent_map(&self, widget: &Self::Type) {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkWidgetClass;
+            if let Some(f) = (*parent_class).map {
+                f(widget.unsafe_cast_ref::<Widget>().to_glib_none().0)
+            }
+        }
+    }
+
+    fn parent_measure(
+        &self,
+        widget: &Self::Type,
+        orientation: Orientation,
+        for_size: i32,
+    ) -> (i32, i32, i32, i32) {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkWidgetClass;
+
+            let mut minimum_size = mem::MaybeUninit::uninit();
+            let mut natural_size = mem::MaybeUninit::uninit();
+            let mut minimum_base_size = mem::MaybeUninit::uninit();
+            let mut natural_base_size = mem::MaybeUninit::uninit();
+            if let Some(f) = (*parent_class).measure {
+                f(
+                    widget.unsafe_cast_ref::<Widget>().to_glib_none().0,
+                    orientation.to_glib(),
+                    for_size,
+                    minimum_size.as_mut_ptr(),
+                    natural_size.as_mut_ptr(),
+                    minimum_base_size.as_mut_ptr(),
+                    natural_base_size.as_mut_ptr(),
+                );
+            }
+            (
+                minimum_size.assume_init(),
```

Comment:
This causes UB if `parent_class.measure == NULL`. You need to implement some kind of default handling here (check what the C code does), or panic (if the C code would just crash / assert).

Reply:
There's always a measure implementation (that returns 0, 0, -1, -1) I just missed this

Reply:
Not if you're deriving from a subclass that has set it to `NULL` explicitly :)

Reply:
So I asked ebassi & mclassen about this and a widget that sets the measure to null without setting a layout-manager will just make thing break and it's safe to assume to be always there

Reply:
Yes it's wrong, but you don't want to run into UB just because someone else does something wrong if you can avoid it. Just panic here or initialize the values to zeroes or so :)

Reply:
Ok, this panics now :)